src/toil/test/jobStores/jobStoreTest.py

```diff
@@ -454,17 +457,32 @@ def testImportSharedFile(self, otherCls):
                        otherCls=activeTestClassesByName)
 
         def testImportHttpFile(self):
-            srcUrl, srcHash = ('https://raw.githubusercontent.com/BD2KGenomics/toil'
-                               '/12c711ff91caa5d8fb37a072bad376f447faa797/Makefile',
-                               '16a38fd7ae233616083d43d81159d4e8')
-            with self.master.readFileStream(self.master.importFile(srcUrl)) as f:
-                self.assertEqual(hashlib.md5(f.read()).hexdigest(), srcHash)
+            httpd = SocketServer.TCPServer(('', 0), StubHttpServer)
+            httpdThread = threading.Thread(target=httpd.handle_request)
```

Comment:
I originally had it that way but changed it to handle one request for simplicity. But I see your point, and you are correct it only requires two changes.